core/src/dp.rs

```diff
@@ -0,0 +1,189 @@
+#[cfg(feature = "test-util")]
+use crate::test_util::dummy_vdaf::Vdaf;
+use anyhow::anyhow;
+use derivative::Derivative;
+#[cfg(feature = "fpvec_bounded_l2")]
+use fixed::traits::Fixed;
+#[cfg(feature = "fpvec_bounded_l2")]
+use prio::flp::{
+    gadgets::{ParallelSumGadget, PolyEval},
+    types::fixedpoint_l2::{compatible_float::CompatibleFloat, FixedPointBoundedL2VecSum},
+};
+use prio::{
+    dp::{
+        distributions::ZCdpDiscreteGaussian, DifferentialPrivacyBudget,
+        DifferentialPrivacyDistribution, DifferentialPrivacyStrategy, DpError,
+    },
+    field::{Field128, Field64},
+    flp::{
+        gadgets::{Mul, ParallelSum, ParallelSumMultithreaded},
+        TypeWithNoise,
+    },
+    vdaf::{xof::XofShake128, AggregatorWithNoise},
+};
+use serde::{Deserialize, Serialize};
+
+// strategy enum for dispatch
+
+#[derive(Debug, Derivative, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub enum DpStrategyInstance {
```

Comment:
Funnily enough it turns out that we don't need this enum anymore, since this functionality is handled on a per-vdaf-basis by the various enums in the `vdaf_instance_strategies` module in `core/vdaf.rs`.